src/engine/source/builder/helperParser.hpp

```diff
@@ -43,67 +59,457 @@ inline std::tuple<std::string, json::Json> toBuilderInput(const HelperToken& hel
 {
-    if (helperToken.args.empty())
+    if (helperToken.args.empty() || helperToken.args[0].empty())
     {
         std::stringstream ss {};
         ss << helperToken;
         throw std::runtime_error(
             fmt::format("Helper {} has no arguments, expected to have target field as first argument", ss.str()));
     }
+    else if (helperToken.args[0][0] != syntax::REFERENCE_ANCHOR)
+    {
+        std::stringstream ss {};
+        ss << helperToken;
+        throw std::runtime_error(fmt::format("Helper {} has no target field as first argument", ss.str()));
+    }
 
+    // Remove reference anchor
     return toBuilderInput({helperToken.name, {helperToken.args.begin() + 1, helperToken.args.end()}},
-                          helperToken.args[0]);
+                          helperToken.args[0].substr(1));
 }
 
-inline std::variant<HelperToken, base::Error> parseHelper(const std::string& text)
+// operators (==, !=, <, >, <=, >=)
+enum class ExpressionOperator
+{
+    EQUAL,
+    NOT_EQUAL,
+    GREATER_THAN,
+    GREATER_THAN_OR_EQUAL,
+    LESS_THAN,
+    LESS_THAN_OR_EQUAL
+};
+
+/**
+ * @brief Token representing an expression
+ *
+ * @details Expression is represented by a field, an operator and a value.
+ *
+ */
+struct ExpressionToken
+{
+    std::string field;
+    ExpressionOperator op;
+    json::Json value;
+};
+
+/**
+ * @brief Obtain builder input parameters from expression token
+ *
+ * @param expressionToken Token representing an expression
+ * @return std::tuple<std::string, json::Json>
+ */
+inline std::tuple<std::string, json::Json> toBuilderInput(const ExpressionToken& expressionToken)
 {
-    static const auto regexPattern = R"(^([\w_]+)\((.*)\)$)";
-    static const re2::RE2 pattern(regexPattern);
+    if (expressionToken.field.empty())
+    {
+        throw std::runtime_error("Expression field is empty");
+    }
 
-    std::string helperName;
-    std::string strArgs;
+    if (expressionToken.op == ExpressionOperator::EQUAL)
+    {
+        return std::make_tuple(expressionToken.field.substr(1), expressionToken.value);
+    }
 
-    HelperToken result;
+    if (expressionToken.op == ExpressionOperator::NOT_EQUAL && !expressionToken.value.isString()
+        && !expressionToken.value.isNumber())
+    {
+        throw std::runtime_error("Not equal operator is not supported for non string or number values");
+    }
 
-    if (re2::RE2::FullMatch(text, pattern, &result.name, &strArgs))
+    // Rest of operators only support string or number values
+    if (!expressionToken.value.isString() && !expressionToken.value.isNumber())
     {
-        if (!strArgs.empty())
+        throw std::runtime_error("Expression value is not string or number");
+    }
+
+    HelperToken helperToken {};
+
+    if (expressionToken.value.isNumber())
+    {
+        helperToken.name = "int";
+        helperToken.args = {std::to_string(expressionToken.value.getInt().value())};
+    }
+    else
+    {
+        helperToken.name = "string";
```

Comment:
Could constants be defined for these literal expressions? What do you think?

Reply:
We may define the names of the helpers as constants, but that's a bigger change not to be tackled on this PR